crates/ruff/src/importer/insertion.rs

```diff
@@ -34,86 +100,174 @@ impl Insertion {
-    pub(super) fn end_of_statement(stmt: &Stmt, locator: &Locator, stylist: &Stylist) -> Insertion {
+    /// in this case is the line after `import math`, and will include a trailing newline.
+    ///
+    /// The statement itself is assumed to be at the top-level of the module.
+    pub(super) fn end_of_statement(
+        stmt: &Stmt,
+        locator: &Locator,
+        stylist: &Stylist,
+    ) -> Insertion<'static> {
         let location = stmt.end();
         let mut tokens =
             lexer::lex_starts_at(locator.after(location), Mode::Module, location).flatten();
         if let Some((Tok::Semi, range)) = tokens.next() {
-            // If the first token after the docstring is a semicolon, insert after the semicolon as an
-            // inline statement;
-            Insertion::new(" ", range.end(), ";")
+            // If the first token after the statement is a semicolon, insert after the semicolon as
+            // an inline statement.
+            Insertion::inline(" ", range.end(), ";")
         } else {
             // Otherwise, insert on the next line.
-            Insertion::new(
+            Insertion::own_line(
                 "",
                 locator.full_line_end(location),
                 stylist.line_ending().as_str(),
             )
         }
     }
 
-    /// Create an [`Insertion`] to insert (e.g.) an import statement at the "top" of a given file,
-    /// along with a prefix and suffix to use for the insertion.
+    /// Create an [`Insertion`] to insert (e.g.) an import statement at the start of a given
+    /// block, along with a prefix and suffix to use for the insertion.
     ///
     /// For example, given the following code:
     ///
     /// ```python
-    /// """Hello, world!"""
-    ///
-    /// import os
+    /// if TYPE_CHECKING:
+    ///     import os
     /// ```
     ///
     /// The insertion returned will begin at the start of the `import os` statement, and will
-    /// include a trailing newline suffix.
-    pub(super) fn start_of_file(body: &[Stmt], locator: &Locator, stylist: &Stylist) -> Insertion {
-        // Skip over any docstrings.
-        let mut location = if let Some(location) = match_docstring_end(body) {
-            // If the first token after the docstring is a semicolon, insert after the semicolon as an
-            // inline statement;
-            let first_token = lexer::lex_starts_at(locator.after(location), Mode::Module, location)
-                .flatten()
-                .next();
-            if let Some((Tok::Semi, range)) = first_token {
-                return Insertion::new(" ", range.end(), ";");
-            }
-
-            // Otherwise, advance to the next row.
-            locator.full_line_end(location)
-        } else {
-            TextSize::default()
-        };
+    /// include a trailing newline.
+    ///
+    /// The block itself is assumed to be at the top-level of the module.
+    pub(super) fn start_of_block(
```

Comment:
I'm trying to figure out the right terminology for "indented" content. In most languages, it's called a block. I, thus far, called it Body in the formatter. 

The only term that I found in the python documentation is *grouping of statements* even though this is, strictly speaking, not correct, because it also applies to non-statements (e.g. match cases)

> Leading whitespace (spaces and tabs) at the beginning of a logical line is used to compute the indentation level of the line, which in turn is used to determine the grouping of statements.